### PR TITLE
Fix a bunch of pylint errors in filtsmooth

### PR DESCRIPTION
**Pull request by pnkraemer**
# In a Nutshell
Fix a bunch of pylint errors in filtsmooth. I refrained from fixing anything that would involve code refactoring, but did all the "quick-and-easy" ones.

### CODE REVIEW

File: src/probnum/filtsmooth/particle/_particle_filter.py [hunk 1]
```diff
@@ -227,7 +237,7 @@ def importance_rv_generator_initial_time(
         for p, w in zip(particles, weights):
             yield importance_rv, dynamics_rv, p, w
 
-    def importance_rv_generator(
+    def _importance_rv_generator(
```
Comment (pnkraemer) [inline src/probnum/filtsmooth/particle/_particle_filter.py:240]: I made these two functions private because they are only used internally anyway.

**Review by marvinpfoertner — CHANGES_REQUESTED** [review]
Please add the colons after argument names without a type annotation again and ignore the resulting pylint error. See the Slack thread in the software channel for more information on this

File: src/probnum/filtsmooth/_timeseriesposterior.py [hunk 2]
```diff
@@ -175,21 +189,25 @@ def sample(
...
 
-        Internally, samples from a base measure are drawn and transformed via self.transform_base_measure_realizations.
+        Internally, samples from a base measure are drawn
+        and transformed via self.transform_base_measure_realizations.
 
         Parameters
         ----------
         rng :
             Random number generator.
         t :
```
Comment (marvinpfoertner) [inline src/probnum/filtsmooth/_timeseriesposterior.py:204]:
```suggestion
        t
```

File: src/probnum/filtsmooth/_timeseriesposterior.py [hunk 3]
```diff
@@ -175,21 +189,25 @@ def sample(
...
-            Locations on which the samples are wanted. Default is none, which implies that
+            Locations on which the samples are wanted.
+            Default is none, which implies that
             self.location is used.
         size :
```
Comment (marvinpfoertner) [inline src/probnum/filtsmooth/_timeseriesposterior.py:208]:
```suggestion
        size
```

File: src/probnum/filtsmooth/_timeseriesposterior.py [hunk 4]
```diff
@@ -209,15 +227,15 @@ def transform_base_measure_realizations(
         base_measure_realizations: np.ndarray,
         t: Optional[ArrayLike],
     ) -> np.ndarray:
-        """Transform a set of realizations from a base measure into realizations from
-        the posterior.
+        """Transform base-measure-realizations into posteriors samples.
 
         Parameters
         ----------
         base_measure_realizations :
```
Comment (marvinpfoertner) [inline src/probnum/filtsmooth/_timeseriesposterior.py:234]:
```suggestion
        base_measure_realizations
```

File: src/probnum/filtsmooth/_timeseriesposterior.py [hunk 5]
```diff
@@ -209,15 +227,15 @@ def transform_base_measure_realizations(
...
             Base measure realizations.
         t :
```
Comment (marvinpfoertner) [inline src/probnum/filtsmooth/_timeseriesposterior.py:236]:
```suggestion
        t
```

File: src/probnum/filtsmooth/particle/_particle_filter.py [hunk 6]
```diff
@@ -52,12 +53,15 @@ class ParticleFilter(_bayesfiltsmooth.BayesFiltSmooth):
     rng :
```
Comment (marvinpfoertner) [inline src/probnum/filtsmooth/particle/_particle_filter.py:53]:
```suggestion
    rng
```

File: src/probnum/filtsmooth/particle/_particle_filter.py [hunk 7]
```diff
@@ -52,12 +53,15 @@ class ParticleFilter(_bayesfiltsmooth.BayesFiltSmooth):
...
         Random number generator.
     with_resampling :
```
Comment (marvinpfoertner) [inline src/probnum/filtsmooth/particle/_particle_filter.py:55]:
```suggestion
    with_resampling
```

File: src/probnum/filtsmooth/particle/_particle_filter.py [hunk 8]
```diff
@@ -52,12 +53,15 @@ class ParticleFilter(_bayesfiltsmooth.BayesFiltSmooth):
...
-        Whether after each step the effective number of particles shall be checked, and, if too low,
+        Whether after each step the effective number of particles
+        shall be checked, and, if too low,
         the state should be resampled. Optional. Default is `True`.
     resampling_percentage_threshold :
```
Comment (marvinpfoertner) [inline src/probnum/filtsmooth/particle/_particle_filter.py:59]:
```suggestion
    resampling_percentage_threshold
```

File: src/probnum/filtsmooth/particle/_particle_filter.py [hunk 9]
```diff
@@ -130,6 +134,11 @@ def filter_generator(
         regression_problem :
```
Comment (marvinpfoertner) [inline src/probnum/filtsmooth/particle/_particle_filter.py:134]:
```suggestion
        regression_problem
```

**Review by marvinpfoertner — APPROVED** [review]

**Pull request merged**